pylabrobot/resources/resource.py

```diff
@@ -136,16 +136,34 @@ def assign_child_resource(self, resource: Resource, location: Optional[Coordinat
     which case you will be responsible for handling any checking, if necessary.
     """
 
-    resource.parent = self
-    resource.location = location
     try:
+      self.check_assignment(resource, location)
+      resource.parent = self
+      resource.location = location
       self.resource_assigned_callback(resource) # call callbacks first.
     except Exception as e:
       resource.parent = None
       resource.location = None
       raise e
 
     self.children.append(resource)
+    
+  def check_assignment(self, resource: Resource, location: Optional[Coordinate]):
+    msgs = []
+
+    # Check for self assignment
+    if resource is self:
+      msgs.append(f"Will not assign resource '{self.name}' to itself.")
+
+    # Check for multiple parents
+    if resource.parent:
```

Comment:
`if resource.parent is not None` is cleaner.

Reply:
Yes :)